### ISSUE TITLE
[Site Editor] Templates and Template Parts screens - Provide a way for users to add a description to  custom templates.

### ISSUE THREAD

**Issue by paaljoachim** [WordPress/gutenberg#36664]
### NB! Refocusing issue from adding a column for published date to instead making it possible for users to add descriptions to custom templates.

-------

The following is based on testing the PR Gutenberg Build from this PR: 
https://github.com/WordPress/gutenberg/pull/36379#issuecomment-973432382

## What problem does this address?

If the user has two similar templates it would be helpful to have a column for when the template was created.

![Screenshot 2021-11-19 at 16 24 43](https://user-images.githubusercontent.com/5323259/142647539-a38fe85b-9f60-49f9-8753-561060ef3fcf.jpg)

## What is your proposed solution?

~~Perhaps something similar to this~~ (Refocusing the issue so users can add a description to custom templates)

![Templates-list-date](https://user-images.githubusercontent.com/5323259/142649096-7d0a2baa-4871-4acb-974a-7ca2a7e7efde.jpg)

**Comment by talldan** [comment]
This isn't part of the design that @jameskoster put together.

At the moment templates and template parts don't support a published date. Like the 'Author' field that would have to be something added to the post type. There's also the same situation with author that a lot of these templates are files, so won't have a concept of being published.

**Comment by jameskoster** [comment]
I wonder if there's a better way to do this. For the vast majority of templates the publish date is not useful information, indeed I'm not sure how we'd even determine the publish date for templates supplied by theme / plugin packages.

> If the user has two similar templates

In this case, they should probably use the description field to elaborate. It's not currently possible to edit the description for custom templates, so perhaps this issue is where we explore how to enable that?

**Comment by paaljoachim** [comment]
I am refocusing this issue. So users can add a description to the custom templates.

**Comment by paaljoachim** [comment]
My initial thought would be using the 3 dot drop down.

![Add-description](https://user-images.githubusercontent.com/5323259/142847677-528bed0a-13cf-4e01-a360-c42244a0976f.jpg)

When there is a description in place "Add description" would change to "Edit description".

**Comment by jameskoster** [comment]
Yeah I think that makes sense. It may also be good to offer a way to add/edit the description whilst editing the template as well:

<img width="661" alt="Screenshot 2021-11-22 at 10 48 14" src="https://user-images.githubusercontent.com/846565/142848349-447033f3-77a3-489b-9309-b264c205364d.png">

**Comment by annezazu** [comment]
Noting that this was [discussed during the 6.0 walkthrough](https://make.wordpress.org/core/2022/03/30/6-0-product-walk-through/) as something to explore.

**Comment by jameskoster** [comment]
We recently [added a 'Summary' panel in the post editor](https://github.com/WordPress/gutenberg/pull/39973) where you can edit the post excerpt. We should explore aligning with that pattern here.
